### PR TITLE
Faerie Dragon species add

### PR DESCRIPTION
**Pull request by RealzHS**

### CODE REVIEW

File: crawl-ref/source/player.cc [hunk 1]
```diff
@@ -5863,7 +5901,7 @@ int player::racial_ac(bool temp) const
     // drac scales suppressed in all serious forms, except dragon
     if (species_is_draconian(species)
         && (!player_is_shapechanged() || form == transformation::dragon
-            || !temp))
+        || !temp))
```
Comment (NormalPerson7) [inline crawl-ref/source/player.cc]: Erm that formatting was correct...

File: crawl-ref/source/player.cc [hunk 2]
```diff
@@ -3167,6 +3179,13 @@ int player_stealth()
             stealth += STEALTH_PIP;
     }
 
+    //Faerie Dragons' bright wings reduce stealth.
+    if (you.species == SP_FAERIE_DRAGON && (you.form == transformation::none
+        || you.form == transformation::appendage))
```
Comment (NormalPerson7) [inline crawl-ref/source/player.cc]: Did you forget blade hands?

File: crawl-ref/source/player.cc (hunk 1 above)
Comment (RealzHS) [reply crawl-ref/source/player.cc]: Fixed.

File: crawl-ref/source/player.cc (hunk 2 above)
Comment (RealzHS) [reply crawl-ref/source/player.cc]: Great catch, I did. And dragon form. Fixed for both.

File: crawl-ref/source/dat/clua/automagic.lua [hunk 3]
```diff
@@ -240,7 +240,11 @@ end
 
 local function mp_is_low()
   local mp, mmp = you.mp()
-  return (100*mp <= AUTOMAGIC_STOP*mmp)
```
Comment (aidanholm) [inline crawl-ref/source/dat/clua/automagic.lua]: Why not use a ternary? I think it makes it clearer:
```lua
local mp, mmp = you.mp()
mp = mp + (you.race() == "Faerie Dragon" and 1 or 0)
return 100*mp <= AUTOMAGIC_STOP*mmp
```
or else just
```lua
if you.race() == "Faerie Dragon" then mp = mp + 1 end
```

File: crawl-ref/source/item-use.cc [hunk 4]
```diff
@@ -701,6 +701,15 @@ bool can_wear_armour(const item_def &item, bool verbose, bool ignore_temporary)
         return false;
     }
 
+    if (you.species == SP_FAERIE_DRAGON && slot == EQ_BODY_ARMOUR)
+    {
+        if (verbose)
+        {
```
Comment (aidanholm) [inline crawl-ref/source/item-use.cc]: No need for the {} around the mprf call here

File: crawl-ref/source/status.cc [hunk 5]
```diff
@@ -929,7 +929,16 @@ static void _describe_airborne(status_info* inf)
     const bool perm      = you.permanent_flight();
     const bool expiring  = (!perm && dur_expiring(DUR_FLIGHT));
     const bool emergency = you.props[EMERGENCY_FLIGHT_KEY].get_bool();
-    const string desc   = you.tengu_flight() ? " quickly and evasively" : "";
+    string desc          = "";
+
+    if (you.tengu_flight())
```
Comment (aidanholm) [inline crawl-ref/source/status.cc]: No need for {} here

File: crawl-ref/source/status.cc [hunk 6]
```diff
@@ -929,7 +929,16 @@ static void _describe_airborne(status_info* inf)
...
+    {
+        desc = " quickly and evasively";
+    }
+    else if (you.faerie_dragon_flight())
```
Comment (aidanholm) [inline crawl-ref/source/status.cc]: ditto

File: crawl-ref/source/tiledoll.cc [hunk 7]
```diff
@@ -286,11 +286,12 @@ void fill_doll_equipment(dolls_data &result)
         tileidx_t ch;
         switch (you.species)
         {
-        case SP_CENTAUR: ch = TILEP_TRAN_STATUE_CENTAUR;  break;
```
Comment (aidanholm) [inline crawl-ref/source/tiledoll.cc]: It's not hugely important, but in general, it's better not to align items like this; it hinders using tools like git-blame. Probably best if you just add the faerie dragon line and let it stick out the side.

File: crawl-ref/source/tiledoll.cc [hunk 8]
```diff
@@ -286,11 +286,12 @@ void fill_doll_equipment(dolls_data &result)
...
-        case SP_NAGA:    ch = TILEP_TRAN_STATUE_NAGA;     break;
-        case SP_FELID:   ch = TILEP_TRAN_STATUE_FELID;    break;
-        case SP_OCTOPODE:ch = TILEP_TRAN_STATUE_OCTOPODE; break;
-        default:         ch = TILEP_TRAN_STATUE_HUMANOID; break;
+        case SP_CENTAUR:       ch = TILEP_TRAN_STATUE_CENTAUR;       break;
```
Comment (aidanholm) [inline crawl-ref/source/tiledoll.cc]: ditto

File: crawl-ref/source/tiledoll.cc [hunk 9]
```diff
@@ -300,11 +301,12 @@ void fill_doll_equipment(dolls_data &result)
     case transformation::lich:
         switch (you.species)
         {
-        case SP_CENTAUR: ch = TILEP_TRAN_LICH_CENTAUR;  break;
-        case SP_NAGA:    ch = TILEP_TRAN_LICH_NAGA;     break;
-        case SP_FELID:   ch = TILEP_TRAN_LICH_FELID;    break;
-        case SP_OCTOPODE:ch = TILEP_TRAN_LICH_OCTOPODE; break;
-        default:         ch = TILEP_TRAN_LICH_HUMANOID; break;
+        case SP_CENTAUR:       ch = TILEP_TRAN_LICH_CENTAUR;       break;
```
Comment (aidanholm) [inline crawl-ref/source/tiledoll.cc]: ditto

File: crawl-ref/source/main.cc [hunk 10]
```diff
@@ -447,6 +447,12 @@ NORETURN static void _launch_game()
 
     _prep_input();
 
+    if (you.species == SP_FAERIE_DRAGON && you.num_turns == 0
```
Comment (aidanholm) [inline crawl-ref/source/main.cc]: These conditionals should be moved into `float_once()`, as the faerie dragon check is currently duplicated. It should also be given a more descriptive name; consider making it static to main.cc and calling it `_faerie_dragon_begin_game_flying()`. You might want to consider using `no_messages` to silence `float_player()`, if the player is supposed to be already flying when they begin the game.

File: crawl-ref/source/player.cc [hunk 11]
```diff
@@ -4869,13 +4887,30 @@ void float_player()
     }
     else if (you.tengu_flight())
         mpr("You swoop lightly up into the air.");
+
```
Comment (aidanholm) [inline crawl-ref/source/player.cc]: No need for the empty lines before/after this case.

**Comment by MarvinPA** [comment]
Closing this, per gammafunk's comment.

**Pull request closed without merging**